pootle/apps/pootle_app/management/commands/refresh_stats.py

```diff
@@ -285,6 +286,12 @@ def _set_wordcount_stats(self, unit_filter):
         key = None
 
         for item in res.iterator():
+            # only include files with ext matching project ext
+            file_ext = os.path.splitext(item['store__name'])[1][1:]
+            proj_ext = item['store__translation_project__project__localfiletype']
+            if file_ext != proj_ext:
+                continue
```

Comment:
I would prefer this filtering to be done in the queryset itself, but when I tried to use an `F` expression it didn't work. Just FYI I was trying the following in the above queryset:

```python
        res = units.filter(store__name__endswith=F('store__translation_project__project__localfiletype')) \
                   .values('store', 'state') \
                   .annotate(wordcount=Sum('source_wordcount')) \
                   .order_by('store', 'state')
```

But I was obviously doing something wrong there.